bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java

```diff
@@ -344,7 +345,32 @@ public void removeHomeGroup(String itemName) {
         itemRegistry.remove(itemName);
     }
 
+    /**
+     * Remove a thing and all its links and items.
+     * If this is a bridge, also remove child things.
+     * @param thingUID
+     */
     public void removeThing(ThingUID thingUID) {
+    	// Lookup the thing in the registry
+    	Thing thing = thingRegistry.get(thingUID);
+    	if(thing == null) {
+    		return;
+    	}
+
+    	// If this is a bridge, remove all child things, their items and links
+        if (thing instanceof Bridge) {
+            Bridge bridge = (Bridge) thing;
+            for (Thing bridgeThing : bridge.getThings()) {
+            	ThingUID bridgeThingUID = bridgeThing.getUID();
+                String bridgeItemName = toItemName(bridgeThingUID);
```

Comment:
I think you can better call 'removeThing(bridgeThingUID)' here. Then it also works recursive with deeper bridge hierarchies.